Final Pokedex Project Submission

Bl4ckBend3r opened the pull request:
This pull request includes the complete Pokedex project: core features, arena mechanics, edit/create functionality, persistent storage via JSON-server, validation using React Hook Form and Zod, as well as the final README documentation.

Bl4ckBend3r committed:
Initial layout + navbar, dynamic Pokemon list, search & pagination

Bl4ckBend3r committed:
Dodano formularz logowania i rejestracji z walidacją i obsługą JSON-servera

Bl4ckBend3r committed:
Dodano stronę ulubionych pokemonów i routing

Bl4ckBend3r committed:
Aktualizacja: arena, edycja, logika walki, ulubione

Bl4ckBend3r committed:
Final project commit: complete implementation with README and production-ready structure

RafalKielbasa commented on README.md:
````diff
@@ -1,12 +1,82 @@
+2. Uruchom serwer danych (JSON Server):
+
+```bash
+npm run server
````
This command is not in the package.json

RafalKielbasa commented on src/subpages/arena/Arena.jsx:
Such extensive logic should be moved to a custom hook

RafalKielbasa commented on src/subpages/arena/Arena.jsx:
```diff
@@ -0,0 +1,263 @@
+  const [showModal, setShowModal] = useState(false);
+  const [isDraw, setIsDraw] = useState(false);
+
+  useEffect(() => {
```
We write useEffects at the bottom of the component.

RafalKielbasa commented on src/subpages/details/PokemonDetails.jsx:
Such extensive logic should be moved to a custom hook

Bl4ckBend3r committed:
Refactor: Zintegrowano wspólny stan kolekcji Pokémonów z hookiem usePokemonCollection

- Stworzono reużywalny hook usePokemonCollection do wspólnego pobierania danych z PokeAPI oraz JSON-servera
- Komponenty Home.jsx, Edit.jsx, Ranking.jsx, Favorites.jsx korzystają teraz z jednego źródła danych
- Nowo stworzone Pokémony pojawiają się poprawnie na liście, w edycji i w rankingu
- Usunięto lokalne useEffect i zduplikowane fetchowanie danych
- Favorites.jsx nie próbuje już pobierać danych stworzonych Pokémonów z PokeAPI (zapobiega 404)
- Poprawiono obsługę ID typu string, np. dla Pokémonów lokalnych (d713), w szczególności w arenie (usunięcie błędu PATCH NaN)
- Cała aplikacja oparta na jednej, spójnej kolekcji Pokémonów, zgodnie z wymaganiami zaliczeniowymi
